kernel/src/utilities/helpers.rs

```diff
@@ -42,3 +42,23 @@ macro_rules! count_expressions {
     ($head:expr $(,)?) => (1usize);
     ($head:expr, $($tail:expr),* $(,)?) => (1usize + count_expressions!($($tail),*));
 }
+
+/// Compute a POSIX-style CRC32 checksum of a slice.
+///
+/// Online calculator: <https://crccalc.com/>
+pub fn crc32_posix(b: &[u8]) -> u32 {
```

Comment:
I think we should add unit tests. Otherwise this looks good to me

Reply:
I agree unit tests would be nice, but this seems like a function of relatively low complexity. I wouldn't want to block this PR on that.